app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java

```diff
@@ -201,22 +202,28 @@ public void setText(String text) {
         }
 
         private void appendToExistingDocument(final File file, final String separator, final boolean showEditor) {
-            Bundle args = new Bundle();
+            final Bundle args = new Bundle();
             args.putSerializable(Document.EXTRA_PATH, file);
             args.putBoolean(Document.EXTRA_PATH_IS_FOLDER, false);
             final Context context = getContext();
             final Document document = Document.fromArguments(context, args);
-            String trimmedContent = document.loadContent(context).trim();
-            String currentContent = TextUtils.isEmpty(trimmedContent) ? "" : (trimmedContent + "\n");
-            document.saveContent(context, currentContent + separator + _sharedText);
-            document.resetModTime();
-
-            if (showEditor) {
-                showInDocumentActivity(document);
-            }
+            String content = document.loadContent(context);
+
+            if (content != null) { // Don't do anything if could not load file
+                content = content.trim();
+                content += (content.isEmpty() ? "" : "\n") + separator + _sharedText;
+                document.saveContent(context, content);
+                document.resetModTime(); // Force load of content in document
+
+                if (showEditor) {
+                    // Open at line
+                    document.setInitialLineNumber(StringUtils.getLineOffsetFromIndex(content, content.length() - 1)[0]);
```

Comment:
maybe INT_MAX  works, so editor just jumps to the end of the file...without calculation